src/west/util.py

```diff
@@ -111,7 +135,7 @@ def west_topdir(start=None, fall_back=True):
     # If you change this function, make sure to update the bootstrap
     # script's find_west_topdir().
 
-    cur_dir = start or os.getcwd()
+    cur_dir = start or get_pwd_or_cwd()
```

Comment:
This isn't the only place where we use os.getcwd(), even outside of the test case code.

I know you're still working on this and that it's just a draft, so I assume there will be test cases once you're ready for this to get merged.

Reply:
Right, as explained in the commit message this place was enough to fix the _limited_ (but useful!) test coverage I tried at the time.
It wouldn't hurt to  fix this problem "incrementally" because it's not like `realpath` or its equivalent is consistently used across the whole zephyr build right now, if you look beyond `west` the situation is already and vastly `abspath/realpath` inconsistent = how I found this issue in the first place.